rocAL/rocAL_pybind/examples/rocAL_api_video_pipeline.py

```diff
@@ -54,35 +55,38 @@ def __init__(self, pipelines, tensor_layout=types.NCHW, reverse_channels=False,
         self.reverse_channels = reverse_channels
         self.tensor_dtype = tensor_dtype
         self.batch_size = self.loader._batch_size
-        self.w = self.loader.getOutputWidth()
-        self.h = self.loader.getOutputHeight()
-        self.n = self.loader.getOutputImageCount()
         self.rim = self.loader.getRemainingImages()
         self.display = display
         self.iter_num = 0
         self.sequence_length = sequence_length
         print("____________REMAINING IMAGES____________:", self.rim)
-        color_format = self.loader.getOutputColorFormat()
-        self.p = (1 if color_format is types.GRAY else 3)
-        self.out = np.empty(
-                (self.batch_size*self.n,int(self.h/self.batch_size), self.w,self.p), dtype="ubyte")
+        self.output = self.dimensions = self.dtype = None
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        self.iter_num +=1
-        if(self.loader.isEmpty()):
+        if (self.loader.isEmpty()):
             raise StopIteration
-        if self.loader.run() != 0:
+
+        if self.loader.rocalRun() != 0:
             raise StopIteration
-        #Copy output from buffer to numpy array
-        self.loader.copyImage(self.out)
-        img = torch.from_numpy(self.out)
-        #Display Frames in a video sequence
+        else:
```

Comment:
I think we should remove this else condition from here.

If Stop Iteration is not raised, we would anyways call getOutputTensors()

Reply:
I'm following the iterator template where we call the if statement for rocalRun and raise stopiteration else we get the output tensors. If this change is needed, do u want to change this in all the iterators as well?

Reply:
Yes, we can change in all iterators.
Test its working